src/main/java/app/model/agents/AgentImp.java

```diff
@@ -120,33 +121,16 @@ public void draw(GraphicsContext gc)
         gc.strokeOval(x , y, radius, radius);
     }
 
-    public boolean agentDetected(Type agentType)
-    {
-        for(Ray r : view)
-        {
-            if(r.getType()==agentType)
-            {
-                return true;
-            }
-        }
-        return false;
-    }
-
-    public Vector closestAgent(Type agentType)
+    public Vector closestTypeSeen(Type type)
```

Comment:
I'm worried that this will return the closest type and not the closest AgentType, which means that we will always get a boundary returned if it's next to a wall.  This method is specifically for detecting the nearest agent in order to conduct state changes

Reply:
AgentType isn't about being Intruder or Guard though, it's about whether it's Target, ACO, WF, etc. We always use ACO as guard and WF as intruder now, but that won't always be the case if we're going to add new agents that can be either.

Type, on the other hand, can only be Intruder or Guard for agents. The rays don't store AgentType either, just Type. So it won't be able to use the method in regards to vision, so not at all.
Unless we make changes to AgentType and store it in Ray too or make a certain AgentType always intruder/guard.

Reply:
And if you give Type.INTRUDER or Type.GUARD (or any other type), it will only consider Rays with that specific type. All others don't get through the if-statement